datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,397 @@
+    imo_num = data.pop("IMO")
+    description = data.pop("Category")
+    flags_former = data.pop("Flags (former)")
+
+    vessel = context.make("Vessel")
+    vessel.id = context.make_id(name, imo_num)
+    vessel.add("name", name)
+    vessel.add("imoNumber", imo_num)
+    vessel.add("type", type)
+    vessel.add("description", description)
+    vessel.add("description", data.pop("Vessel information"))
+    vessel.add("callSign", data.pop("Call sign"))
+    vessel.add("flag", data.pop("Flag (Current)"))
+    vessel.add("mmsi", data.pop("MMSI"))
+    vessel.add("buildDate", data.pop("Build year"))
+    for web_resource in data.pop("Web resources"):
+        vessel.add("sourceUrl", web_resource)
+    for name in h.multi_split(data.pop("Former ship names"), [" / "]):
+        vessel.add("previousName", name)
+    for flag in h.multi_split(flags_former, [" / "]):
+        vessel.add("pastFlags", flag)
+    vessel.add("topics", "poi")
+
+    sanction = h.make_sanction(context, vessel)
+    sanction.add("country", data.pop("Sanctions", None))
+    sanction.add("sourceUrl", link)
+
+    context.emit(vessel, target=True)
+    context.emit(sanction)
+
+    linked_entity_name = data.pop(
+        "The person in connection with whom sanctions have been applied"
+    )
+    if linked_entity_name != "":
+        linked_entity = context.make("LegalEntity")
+        linked_entity.id = context.make_id(linked_entity_name)
+        linked_entity.add("name", linked_entity_name)
+        linked_entity.add("topics", "poi")
+        context.emit(linked_entity, target=True)
+
+    com_manager = data.pop("Commercial ship manager (IMO / Country / Date)")
+    if com_manager != "":
+        com_manager_parts = com_manager.split(" / ")
+        if len(com_manager_parts) == 3:
+            com_name_imo, com_country, com_date = com_manager_parts
+            if len(com_name_imo.split(" (")) == 2:
+                com_name, com_imo = com_name_imo.split(" (")
+            else:
+                overrides = lookup_override(context, com_name_imo, "com_manager")
+                com_name = overrides.get("name")
+                com_imo = overrides.get("registrationCode")
+
+            com_manager = context.make("LegalEntity")
+            com_manager.id = context.make_id(com_name, com_country)
+            com_manager.add("name", com_name)
+            com_manager.add("registrationNumber", com_imo)
+            com_manager.add("country", com_country)
+            com_manager.add("topics", "poi")
+            context.emit(com_manager, target=True)
+
+            com_rep = context.make("Representation")
+            com_rep.id = context.make_id(
+                vessel.id, "commercially managed by", com_manager.id
+            )
+            com_rep.add("client", vessel.id)
+            com_rep.add("agent", com_manager.id)
+            com_rep.add("role", "Commercial ship manager")
+            h.apply_date(com_rep, "startDate", com_date)
+
+            context.emit(com_rep)
```

Comment:
this looks ripe for factoring out to a dedicated function `crawl_ship_manager(context, vessel, com_manager)`